src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.Eventing;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace WebJobs.Script.WebHost.Diagnostics
+{
+    public static class MetricsEventManager
+    {        
+        private static FunctionActivityTracker instance = null;
+
+        private static object functionActivityTrackerLockObject = new object();
+
+        public static void BeginFunctionEventHandler()
+        {
+            lock (functionActivityTrackerLockObject)
```

Comment:
We need a way to serialize these events so that we can find out when the first function starts and when the last function finishes. All operations performed inside the lock are in memory so it should be pretty fast. Do you have any specific perf concerns?